lib/byebug/breakpoint.rb

```diff
@@ -31,12 +31,13 @@ def self.add(file, line, expr = nil)
     end
 
     #
-    # Removes a breakpoint
+    # Removes a breakpoint and returns the breakpoint information
     #
     # @param id [integer] breakpoint number
     #
     def self.remove(id)
-      Byebug.breakpoints.reject! { |b| b.id == id }
+      bi = Byebug.breakpoints.find_index { |b| b.id == id }
+      bi ? Byebug.breakpoints.delete_at(bi) : nil
```

Comment:
Well, actually the other thing this does is give the caller status as to whether it found the breakpoint to remove or not. So _some_ change is desirable. 

If you view lib/byebug/breakpoint.rb as an API interface returning the information about what was removed is more useful. However if you are view this as just servicing the delete command as it is written, then assuming you address the status aspect, then sure change it.